serviceinfo-api-server/src/main.rs

```diff
@@ -224,7 +227,64 @@ async fn serviceinfo_handler(
         .modules
         .contains(&FedoraIotServiceInfoModule::SSHKey.into())
     {
-        if let Some(initial_user) = &user_data.service_info_configuration.settings.initial_user {
+        //here we first check if the reuested device has device-specific file stored
+        //in device_specific_store_driver ,if not this device gets config from base file
+        let settings: ServiceInfoApiServerSettings = match settings_for("serviceinfo-api-server") {
+            Ok(config) => config.try_deserialize(),
+            Err(e) => {
+                return Err(warp::reject::custom(RtrFailure(e)));
+            }
+        }
+        .map_err(|e| warp::reject::custom(RtrFailure(e.into())))?;
+
+        let path_per_device_store = match settings.device_specific_store_driver {
+            StoreConfig::Directory { mut path } => {
+                let file_name = format!(
+                    "{}.yml",
+                    query_info.device_guid.to_string().replace('\"', "")
+                );
+                path.push(file_name);
+                path.to_string_lossy().into_owned()
+            }
+        };
+
+        if Path::new(&path_per_device_store).exists() {
+            //device specific config
+            let per_device_settings = match settings_per_device(&path_per_device_store) {
+                Ok(config) => config.try_deserialize::<ServiceInfoSettings>(),
+                Err(e) => {
+                    return Err(warp::reject::custom(RtrFailure(e)));
```

Comment:
Since we are returning an error to the client, if the file does not exist or something similar, we do not want to leak the path were the things are stored in the server, so I would log the error in the server and return a more generic error message to the client.

Reply:
I would just log the error at warning at least and avoid returning the error